### PR TITLE
Set state overwrite

### PR DESCRIPTION
**Pull request by miiu96**
- Added a new endpoint `simulator/set-state-overwrite` that is similar with the `set-state` endpoint,  in addition, it will clean the old state of the provided addresses
- Changed the version of the `mx-chain-go` module

### CODE REVIEW

**Review by iulianpascalau — DISMISSED** [review]

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/multiversx/mx-chain-core-go v1.2.19
 	github.com/multiversx/mx-chain-go v1.7.9-0.20240424111810-ebb386a3356d
 	github.com/multiversx/mx-chain-logger-go v1.0.14
-	github.com/multiversx/mx-chain-proxy-go v1.1.47
+	github.com/multiversx/mx-chain-proxy-go v1.1.48
```
Comment (iulianpascalau) [inline go.mod:10]: mx-chain-go v1.7.9 ?

Comment (miiu96) [reply go.mod:10]: done

**Review by iulianpascalau — DISMISSED** [review]

File: pkg/facade/simulatorFacade.go [hunk 2]
```diff
@@ -48,6 +48,21 @@ func (sf *simulatorFacade) SetStateMultiple(stateSlice []*dtos.AddressState) err
 	return sf.simulator.SetStateMultiple(stateSlice)
 }
 
+// SetStateMultipleOverwrite will set the entire state for the provided address and cleanup the old state of the provided addresses
+func (sf *simulatorFacade) SetStateMultipleOverwrite(stateSlice []*dtos.AddressState) error {
+	accounts := make([]string, 0)
```
Comment (ssd04) [inline pkg/facade/simulatorFacade.go:53]:
```suggestion
	accounts := make([]string, 0, len(stateSlice))
```
?

File: pkg/proxy/api/endpoints.go [hunk 3]
```diff
@@ -142,6 +144,23 @@ func (ep *endpointsProcessor) setStateMultiple(c *gin.Context) {
...
+	var stateSlice []*dtos.AddressState
+	err := c.ShouldBindJSON(&stateSlice)
+	if err != nil {
+		shared.RespondWithBadRequest(c, fmt.Sprintf("invalid state structure, error: %s", err.Error()))
+		return
+	}
+
+	err = ep.facade.SetStateMultipleOverwrite(stateSlice)
+	if err != nil {
+		shared.RespondWithBadRequest(c, fmt.Sprintf("cannot set state, error: %s", err.Error()))
```
Comment (ssd04) [inline pkg/proxy/api/endpoints.go:157]:
```suggestion
		shared.RespondWithBadRequest(c, fmt.Sprintf("cannot overwrite state, error: %s", err.Error()))
```
or something similar to defferentiate from simple set state

File: pkg/facade/simulatorFacade.go (hunk 2 above)
Comment (miiu96) [reply pkg/facade/simulatorFacade.go:53]: done

File: pkg/proxy/api/endpoints.go (hunk 3 above)
Comment (miiu96) [reply pkg/proxy/api/endpoints.go:157]: done

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**